### DISCUSSION TITLE
The foreground color is not rendering as expected

### DISCUSSION THREAD

**Discussion by eatkins** [kovidgoyal/kitty#7509] [Q&A] [answered]
I am running the latest version of kitty on mac os and I am finding that kitty is not rendering the foreground color that I have set correctly. In an empty kitty.conf I have set `foreground   #00FEA7` but when I measure the font color with the built in digital color meter, it is rendering as #74feae. Any thoughts as to why I am seeing a discrepancy? I have attached two screenshots. One is the kitty.conf file rendered with the built in terminal app, the other with kitty. EDIT: I noticed I have the same problem of unwanted red if I try to set the background color.

Kitty:
<img width="567" alt="Screen Shot 2024-06-04 at 11 41 58 AM" src="https://github.com/kovidgoyal/kitty/assets/2658825/a2d41daf-7b9b-4529-9dc7-06a3ecee4911">

Terminal.app:
<img width="604" alt="Screen Shot 2024-06-04 at 11 43 02 AM" src="https://github.com/kovidgoyal/kitty/assets/2658825/11fb6748-1d9a-47e0-9e50-5a0abe675ebe">

**Comment by kovidgoyal** [comment]
You are rendering it with kitty you are rendering it with vim or nvim
or something similar. The color comes from the editors color theme, not
kitty.

**Comment by eatkins** [comment]
The unwanted red values show up in any program. It just happened to be the case that I took the screenshots in nvim. I could reproduce the exact same issue in an empty bash, zsh or fish shell prompt.

**Reply by eatkins** [reply]
I was curious and I ran kitty under virtualbox in Ubuntu and the colors do render correctly there so I only see this behavior when running under macos.

**Comment by kovidgoyal** [comment, marked as answer]
If you want to test colors, you do it via printf not some third party
program. See
https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_(Select_Graphic_Rendition)_parameters

And if you are talking about how the colors are interpreted, see
https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.macos_colorspace

**Comment by eatkins** [comment]
Yes, the problem was that I was measuring the native values with digital color meter (which is built into mac os, not third party). I switched to macos_colorspace default and now the colors render as expected.
